### PR TITLE
progress bar for add-on uploads

### PR DESCRIPTION
**Pull request by ngokevin**
- Uses ```superagent's .on('progress`)```
- Fixes file size display, was incorrectly representing bytes as KB

screen shot 2015-09-11 at 4 25 22 pm
screen shot 2015-09-11 at 4 19 45 pm

### CODE REVIEW

**Comment by sherlockedci** [comment]
[ The Adventure of the Black Build](http://sherlocked.dev.mozaws.net/builds/79942366)

**Comment by sherlockedci** [comment]
[ The Adventure of the Naval Build](http://sherlocked.dev.mozaws.net/builds/79942546)

File: src/media/js/addon/components/progressBar.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    unit: '%',
+  };
+
+  render() {
+    const barFillStyle = {
+      width: `${this.props.loadedSize / this.props.totalSize * 100}%`
+    };
+
+    return (
+      <div className="progress-bar">
```
Comment (diox) [inline src/media/js/addon/components/progressBar.js]: No love for `<progress>` ?

Comment (ngokevin) [inline src/media/js/addon/components/progressBar.js]: well...i'm embarassed i didn't know about that element

**Comment by sherlockedci** [comment]
[ The Adventure of the Thor Build](http://sherlocked.dev.mozaws.net/builds/79943651)

File: src/media/js/addon/actions/submit.js [hunk 2]
```diff
@@ -13,24 +13,27 @@ import Url from 'urlgray';
...
-export const VALIDATION_FAIL = 'SUBMISSION_ADDON__VALIDATION_FAIL';
+export const VALIDATION_FAIL = 'ADDON_SUBMIT__VALIDATION_FAIL';
 const validationFail = createAction(VALIDATION_FAIL);
 
-export const SUBMIT_OK = 'SUBMISSION_ADDON__SUBMIT_OK';
+export const SUBMIT_OK = 'ADDON_SUBMIT__SUBMIT_OK';
 const submitOk = createAction(SUBMIT_OK);
 
-export const SUBMIT_ERROR = 'SUBMISSION_ADDON__SUBMIT_ERROR';
+export const SUBMIT_ERROR = 'ADDON_SUBMIT__SUBMIT_ERROR';
```
Comment (chuckharmston) [inline src/media/js/addon/actions/submit.js]: Good change.

File: src/media/js/addon/containers/dashboardDetail.js [hunk 3]
```diff
@@ -60,14 +62,13 @@ export class AddonDashboardDetail extends React.Component {
 
 
 export default connect(
-  state => ({
+  state => ({...state.addonSubmitVersion, ...{
     addon: state.addonDashboard.addons[state.router.params.slug],
-    addonSubmitVersion: state.addonSubmitVersion,
     slug: state.router.params.slug,
-  }),
+  }}),
```
Comment (chuckharmston) [inline src/media/js/addon/containers/dashboardDetail.js]: Did not realize you could do that. This is very cool.

File: src/media/js/addon/containers/submit.js [hunk 4]
```diff
@@ -16,8 +16,10 @@ export class AddonSubmit extends React.Component {
 
   static propTypes = {
     isSubmitting: React.PropTypes.bool,
+    uploadLoaded: React.PropTypes.number,
+    uploadTotal: React.PropTypes.number,
+    validationErrorMessage: React.PropTypes.string,
     submit: React.PropTypes.func.isRequired,
```
Comment (chuckharmston) [inline src/media/js/addon/containers/submit.js]: :fries:

**Comment by chuckharmston** [comment]
Sherlocked is failing.

**Comment by chuckharmston** [comment]
Looks great, r+wc

File: src/media/js/addon/components/progressBar.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        <div className="progress-bar-bar">
+          <div className="progress-bar-fill" style={barFillStyle}/>
+        </div>
+
+        <span className="progress-bar-text">
+          {this.props.loadedSize}{this.props.unit} /
```
Comment (chuckharmston) [inline src/media/js/addon/components/progressBar.js]: Maybe we should round these numbers. 2 decimal points is more than enough.

**Comment by ngokevin** [comment]
It only fails in the push build for some reason (for me), it works in the PR build

File: src/media/js/addon/components/upload.js [hunk 6]
```diff
@@ -57,21 +61,33 @@ export default class AddonUpload extends React.Component {
...
           <label htmlFor="submission-addon--zip">Add-on ZIP File:</label>
+
           <FileReaderInput as="buffer" accept=".zip"
                            id="submission-addon--zip"
                            onChange={this.handleChange}>
             <div className="form-inline--file-input"
                  data-file-input--has-data={!!this.state.fileName}>
               {this.state.fileName ?
-               `${this.state.fileName} (${this.state.fileSize}KB)` :
+               `${this.state.fileName} (${this.state.fileSize / 1000000}MB)` :
```
Comment (chuckharmston) [inline src/media/js/addon/components/upload.js]: 1048576

**Pull request merged**